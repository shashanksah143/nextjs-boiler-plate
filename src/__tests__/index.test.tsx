import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Home from "../pages/index";

describe("Home Page", () => {
  it("renders the main heading", () => {
    render(<Home />);

    const heading = screen.getByRole("heading", {
      name: /hello world/i,
    });

    expect(heading).toBeInTheDocument();
  });

  it("has the correct document structure", () => {
    render(<Home />);

    // Check that the main container is present
    const mainDiv = screen.getByRole("heading").closest("div");
    expect(mainDiv).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = render(<Home />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
