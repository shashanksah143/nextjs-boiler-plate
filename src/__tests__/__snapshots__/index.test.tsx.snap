// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page matches snapshot 1`] = `
<div>
  <h1>
    Hello world
  </h1>
</div>
`;
